internal/hardware/hardware.go

```diff
@@ -108,3 +125,83 @@ func GetMutableHardwareInfoDelta(hardwareMutableInfoPrevious models.HardwareInfo
 
 	return hardwareInfo
 }
+
+// GetDevices recursively traverses a directory specified by path
+// and returns all devices found there.
+// Shamelessly copied from 	https://github.com/opencontainers/runc/blob/main/libcontainer/devices/device_unix.go
```

Comment:
what's the penalty for adding it as a dependency and consume it directly? (`GetDevices` is an exported function)

Reply:
I've modified it a little bit to comply with our needs but I'm looking into it. maybe we can consume it as it is.

Reply:
done